library/Translate/src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -265,47 +286,26 @@ populateModMap = do
       -- these markings as they are no longer useful
       -- TODO ∷
       -- should we even append modM to this
-      put @"modMap" (modM <> Map.fromList (concatMap f opens))
+      assocNameWithAlias <- concatMapM f opens
+      put @"modMap" (modM <> Map.fromList assocNameWithAlias)
       where
-        f y = catMaybes (fmap addNewSymbol openList)
+        f y = do
+          openList <- inScopeNames nameSpace
+          maybeAssocNameWithAlias <- traverse addNewSymbol openList
+          pure (catMaybes maybeAssocNameWithAlias)
```

Comment:
Due to accessing information is now an effect, we have to promote our `concatMap` to `concatMapM`. However the code is a lot cleaner to read now